src/r1cs/constraint_system.rs

```diff
@@ -53,19 +53,17 @@ pub trait ConstraintSystem {
     /// ```
     fn constrain(&mut self, lc: LinearCombination);
 
-    /// Obtain a challenge scalar bound to the assignments of all of
-    /// the externally committed wires.
+    /// Randomize the constraints using a randomly sampled challenge scalar
+    /// bound to the assignments of the non-randomized variables.
```

Comment:
Saying "randomize the constraints" sounds like there are some existing constraints, and then, these existing constraints are randomized, when we really mean that this is a hook for adding constraints which depend on a random value.

I'm not sure exactly what would be better, but maybe this could be explained in terms of the role of the callback?  Right now the role of the `Fn(&mut Self, Scalar)` is a little unclear. We should probably say that the callback is supposed to use the random input scalar to construct further constraints.

Also, is there a more precise name for what we're doing than `randomized_constraints`? It feels like it's still not quite right, because it's not that we're adding randomized constraints in *this* call, which is a hook to call the closure to add them later.

Reply:
`define_randomized_constraints`?

Reply:
Renamed to `specify_randomized_constraints` and added more API documentation in cced8b7